libgnucash/engine/gnc-pricedb.cpp

```diff
@@ -690,14 +690,14 @@ compare_prices_by_date(gconstpointer a, gconstpointer b)
                          gnc_price_get_guid((GNCPrice *) b));
 }
 
-static bool
-price_is_duplicate (const GNCPrice *pPrice, const GNCPrice *cPrice)
+static int
+price_compare (const GNCPrice *pPrice, const GNCPrice *cPrice)
```

Comment:
Keep the old name. This isn't a real `GCompareFunc`, it can't return -1 because of the use of `operator||` so it can't be used for sorting.

You should also fix the variable names (p_price instead of pPrice) to comply with the coding standard, and since a price list should have only prices between identical commodity/currency pairs and should contain only one price per date I'd check date, value, commodity, then currency.